File: 242-ValidAnagram/242-ValidAnagram.java
// Last updated: 6/12/2025, 8:30:07 AM
class Solution {
    public boolean isAnagram(String s, String t) {
        char[] s1 = s.toCharArray();
        char[] s2 = t.toCharArray();
        Arrays.sort(s1);
        Arrays.sort(s2);
        if(s1.length != s2.length) return false;
        for(int i=0; i<s1.length; i++){
            if(s1[i]!=s2[i]) return false;
        }
        return true;
    }
}

File: 242-ValidAnagram/242-ValidAnagram.py
# Last updated: 6/12/2025, 8:30:10 AM
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        # return sorted(s) == sorted(t)
        sCounts = {}
        tCounts = {}
        if len(s) != len(t):
            return False
        for i in range(len(s)):
            sCounts[s[i]] = sCounts.get(s[i], 0) + 1
            tCounts[t[i]] = tCounts.get(t[i], 0) + 1
        if len(sCounts) != len(tCounts):
            return False
        for c in sCounts.keys():
            if tCounts.get(c) != sCounts[c]:
                return False
        return True